treewide: Add missing PKG_MAINTAINER

stokito opened the pull request:
The luci-app-bcp38 had a maintainer from luci-app-acme and that looks like a copy-paste error. So it's maintainer was changed to an author @danrl

Daniel F. Dickinson changed email address to <dfdpublic@wildtechgarden.ca>

luci-all-lxl has a maintainer Petar Koretic <petar.koretic@sartura.hr> but there is no corresponding GitHub account. So Dirk Brenken was added as a second maintainer: he answered on an issue of the app.

When maintainer wasn't set the initial author was used, or most contributor or Jo-Philipp Wich as a default.

hnyman commented:
The bcp38 maintainer change is wrong. @tohojo is the author and maintainer of that app.
(You were probably misled by the 7-year ago move from packages repo, if you did not follow the whole commit and PR chain.)

https://github.com/openwrt/packages/pull/4013

https://github.com/openwrt/packages/commits/c8add3368b8bd7767bbd540539dcd4a97112b4e1/net/luci-app-bcp38/Makefile

stokito commented:
@hnyman thank you for the clarification. Reverted that change.

@feckert I added missed PKG_LICENSE. Please review last two commits.

There is a problem that the `example` app has a GPLv2 copyright and when a new app is created it's copied too. This creates a problem if an author didn't specified the PKG_LICENSE directly.

E.g. for the luci-app-usteer I had to specify the PKG_LICENSE=GPL-2.0 but I'm pretty sure that its author @Ramon00 would be fine with Apache-2.0. @Ramon00 could you please clarify which license you used for your app? Can I set it to the Apache-2.0 that most of Luci code use?

To avoid such problems I would like to propose to change the luci-app-example to Apache-2.0

Ramon00 commented:
@stokito Im fine with that

stokito commented:
Thank you. I changed the app-useer to Apache-2.0 and force pushed.

Ramon00 commented on applications/luci-app-usteer/Makefile:
can you update my email to: Ramon00c00@gmail.com

stokito replied:
fixed in cfae1f9

hnyman commented:
One comment to the syntax: I think that in case of duplicate maintainers, the list should be comma-separated so that it would be acceptable to email clients as recipients list.

systemcrash commented:
Agreed. Comma simplest. Although it does not yet help (multiple authors case) the new pkg search that @spooren made.

stokito commented:
> in case of duplicate maintainers, the list should be comma-separate

From what I saw in all places maintainers are separated by a new line OR by a comma if written in one line. I just followed the common style. If we want to separate by a coma and new line then we should apply the same style for all makefiles. We can do that in a separate PR.

The PR can be merged as is.
If anything should be changed please let me know and I'll update the PR

systemcrash commented:
@hnyman do you have any further comments?
@stokito rebase please

stokito commented:
@systemcrash rebased

stokito commented:
@hnyman thank you, I added the commas in a separate commit.

> changing from GPL2 to Apache without explicit consent from the author

For each such change I checked the license declared in file headers inside. We have the luci-example-app that has the GPLv2 and this was copied without noticing.

> some unrelated changes, like changing a few mail addresses in the copyrights

Only one email was changed from <openwrt@daniel.thecshore.com> to <dfdpublic@wildtechgarden.ca> which is a public email on his [github](https://github.com/danielfdickinson). The old domain daniel.thecshore.com doesn't have MX records. The change is minor and I made it when looked for his github account for the CODEOWNERS.

So from my side everything looks fine now.

stokito commented:
Ok, so can we merge this? The change shouldn't break anything

systemcrash commented:
Looks largely OK - although my p910nd rewrite seemed to acquire jow as a maintainer. Logic? Not against it - just want to understand the decision making here.

LICENSE changes which specifically point to the main luci repo license are explicit and OK.

stokito commented:
When I created the codeowners and filled missed maintainers you didn't rewrote it yet.
Then I added you and kept jow.

2 мая 2024 г. 02:16:31 Paul Donald ***@***.***>:

> 
> Looks largely OK - although my p910nd rewrite seemed to acquire jow as a maintainer. Logic? Not against it - just want to understand the decision making here.
> 
> LICENSE changes which specifically point to the main luci repo license are explicit and OK.
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/openwrt/luci/pull/7023#issuecomment-2089275687], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AADFODUX4FMVACTKKKR5TUTZAFZU7AVCNFSM6AAAAABFFU3MHGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOBZGI3TKNRYG4].
> You are receiving this because you were mentioned.
> [Отслеживающее изображение][https://github.com/notifications/beacon/AADFODRYIU6NN4FS3MUZBXDZAFZU7A5CNFSM6AAAAABFFU3MHGWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTT4Q7ISO.gif]
>
